soc/arm/silabs_exx32/common/soc.c

```diff
@@ -60,9 +60,16 @@ static ALWAYS_INLINE void clock_init(void)
 	CMU_OscillatorEnable(cmuOsc_HFRCO, false, false);
 #elif (defined CONFIG_CMU_HFCLK_HFRCO)
 	/*
-	 * This is the default clock, the controller starts with, so nothing to
-	 * do here.
+	 * This is the default clock, the controller starts with
 	 */
+
+	if (CONFIG_CMU_HFRCO_FREQ) {
+		/* Setting system HFRCO frequency */
+		CMU_HFRCOBandSet(CONFIG_CMU_HFRCO_FREQ);
```

Comment:
This call will not work on efm32wg and efm32hg series. The function is declared as
```
void                  CMU_HFRCOBandSet(CMU_HFRCOBand_TypeDef band);
```
where 
```
typedef enum {
  cmuHFRCOBand_1MHz  = 0,      /**< 1 MHz HFRCO band  */
  cmuHFRCOBand_7MHz  = 1,      /**< 7 MHz HFRCO band  */
  cmuHFRCOBand_11MHz = 2,     /**< 11 MHz HFRCO band */
  cmuHFRCOBand_14MHz = 3,     /**< 14 MHz HFRCO band */
  cmuHFRCOBand_21MHz = 4,     /**< 21 MHz HFRCO band */
} CMU_HFRCOBand_TypeDef;
```
Also, in this case `0` is a valid value.